From 2e2db9731d22917135bdf86519094e24c3145562 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Tue, 14 May 2024 17:37:20 +0200
Subject: [PATCH] libtoolize: Don't use uninitialized variable.

* libtoolize.in (func_serial_update): Initialize my_src_serial and
my_dest_serial before use.
* tests/libtoolize.at: Update expected test result.
---
 libtoolize.in       | 4 +++-
 tests/libtoolize.at | 3 ---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/libtoolize.in b/libtoolize.in
index 336fc81b..23741aaf 100644
--- a/libtoolize.in
+++ b/libtoolize.in
@@ -657,6 +657,8 @@ func_serial_update ()
 
     test -f "$my_srcfile" || func_fatal_error "'$my_srcfile' does not exist."
 
+    my_src_serial=
+    my_dest_serial=
     if test -f "$my_destfile"; then
       my_src_serial=`func_serial "$my_srcfile" "$my_macro_regex"`
       my_dest_serial=`func_serial "$my_destfile" "$my_macro_regex"`
@@ -696,7 +698,7 @@ func_serial_update ()
       # serial tags, so the update message will be correctly given
       # if aclocal.m4 contains an untagged --i.e older-- macro file):
       *)
-        if test -f aclocal.m4; then
+        if test -f aclocal.m4 && test "X$my_src_serial" != X; then
           func_serial_max \
               "$my_src_serial" `func_serial aclocal.m4 "$my_macro_regex"`
           if test "X$my_src_serial" = "X$func_serial_max_result"; then
diff --git a/tests/libtoolize.at b/tests/libtoolize.at
index cbc21d68..a9ed75a6 100644
--- a/tests/libtoolize.at
+++ b/tests/libtoolize.at
@@ -681,11 +681,8 @@ AT_DATA([expout],
 libtoolize: copying file 'm4/libtool.m4'
 libtoolize: copying file 'm4/ltoptions.m4'
 libtoolize: copying file 'm4/ltsugar.m4'
-libtoolize: You should add the contents of 'm4/ltsugar.m4' to 'aclocal.m4'.
 libtoolize: copying file 'm4/ltversion.m4'
-libtoolize: You should add the contents of 'm4/ltversion.m4' to 'aclocal.m4'.
 libtoolize: copying file 'm4/lt~obsolete.m4'
-libtoolize: You should add the contents of 'm4/lt~obsolete.m4' to 'aclocal.m4'.
 ]])
 
 LT_AT_CHECK_LIBTOOLIZE([--copy], 0, expout)
-- 
2.34.1

